### PR TITLE
[ASSIST 397, 393, 395, 401] Caseload Management

### PR DESCRIPTION
**Pull request by mxm1070**
https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/397/
https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/393/
https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/401/
https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/395/

### CODE REVIEW

File: app-modules/caseload-management/database/factories/CaseloadFactory.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+{
+    /**
+     * @return array<string, mixed>
+     */
+    public function definition(): array
+    {
+        return [
+            'name' => fake()->words(asText: true),
+            'model' => fake()->randomElement(CaseloadModel::cases()),
+            'type' => CaseloadType::Dynamic, //TODO: add static later
```
Comment (danharrin) [inline app-modules/caseload-management/database/factories/CaseloadFactory.php:24]: Is static still later?

Comment (mxm1070) [reply app-modules/caseload-management/database/factories/CaseloadFactory.php:24]: I've technically added it, but I've left it disabled until we decide exactly how it's supposed to work.

File: app-modules/assist-data-model/src/Providers/AssistDataModelServiceProvider.php [hunk 2]
```diff
@@ -20,5 +22,12 @@ public function boot(): void
         Relation::morphMap([
             'student' => Student::class,
         ]);
+
+        $this->registerObservers();
+    }
+
+    protected function registerObservers(): void
+    {
+        Caseload::observe(CaseloadObserver::class);
```
Comment (Orrison) [inline app-modules/assist-data-model/src/Providers/AssistDataModelServiceProvider.php:31]: Shouldn't this be in the CaseLoadManagementServiceProvider?

File: app-modules/caseload-management/database/migrations/2023_09_21_142344_create_caseload_subjects_table.php [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        Schema::create('caseload_subjects', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+
+            $table->string('subject_id');
+            $table->string('subject_type');
+
+            $table->foreignUuid('caseload_id')->constrained('caseloads');
+
+            $table->unique(['subject_id', 'subject_type', 'caseload_id']);
```
Comment (Orrison) [inline app-modules/caseload-management/database/migrations/2023_09_21_142344_create_caseload_subjects_table.php:18]: I don't think this should be unique. A subject could be a part of different Caseloads, I believe.

File: app/Providers/Filament/AdminPanelProvider.php [hunk 4]
```diff
@@ -49,6 +49,8 @@ public function panel(Panel $panel): Panel
                 'primary' => Color::hex('#2bb8b3'),
             ])
             ->viteTheme('resources/css/filament/admin/theme.css')
+            ->readOnlyRelationManagersOnResourceViewPagesByDefault(false)
+            // ->maxContentWidth('full')
```
Comment (Orrison) [inline app/Providers/Filament/AdminPanelProvider.php:53]: Let's delete this comment if we are not going to use it here.

**Review by Orrison — CHANGES_REQUESTED** [review]
Just a couple of small changes and then this should be good

File: app-modules/assist-data-model/src/Providers/AssistDataModelServiceProvider.php (hunk 2 above)
Comment (mxm1070) [reply app-modules/assist-data-model/src/Providers/AssistDataModelServiceProvider.php:31]: Whoops!

File: app/Providers/Filament/AdminPanelProvider.php (hunk 4 above)
Comment (mxm1070) [reply app/Providers/Filament/AdminPanelProvider.php:53]: Leaving it pending ticket.

File: app-modules/caseload-management/database/migrations/2023_09_21_142344_create_caseload_subjects_table.php (hunk 3 above)
Comment (mxm1070) [reply app-modules/caseload-management/database/migrations/2023_09_21_142344_create_caseload_subjects_table.php:18]: Removed

File: app-modules/assist-data-model/src/Providers/AssistDataModelServiceProvider.php (hunk 2 above)
Comment (mxm1070) [reply app-modules/assist-data-model/src/Providers/AssistDataModelServiceProvider.php:31]: Moved

File: app-modules/caseload-management/database/migrations/2023_09_21_142344_create_caseload_subjects_table.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        Schema::create('caseload_subjects', function (Blueprint $table) {
+            $table->uuid('id')->primary();
+
+            $table->string('subject_id');
+            $table->string('subject_type');
+
+            $table->foreignUuid('caseload_id')->constrained('caseloads');
+
+            $table->timestamps();
```
Comment (Orrison) [inline app-modules/caseload-management/database/migrations/2023_09_21_142344_create_caseload_subjects_table.php:18]:
```suggestion
            $table->foreignUuid('caseload_id')->constrained('caseloads');
            
            $table->index(['subject_type', 'subject_id']);

            $table->timestamps();
```

Adding an index for the morph here will help some performance.

**Review by Orrison — CHANGES_REQUESTED** [review]
One last quick change.

**Review by Orrison — APPROVED** [review]

**Pull request merged**